### PR TITLE
Sd v2

### PR DESCRIPTION
**Pull request by seandough3rty**

### CODE REVIEW

File: sd/StringParser/src/com/company/FileAdapter.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
+package com.company;
+import java.nio.file.Paths;
+import java.util.*;
+import java.io.*;
+
+public class FileAdapter {
+
+    public static HashSet<String> blackListFromLanguageID(String languageID) throws IOException
```
Comment (waqarmuhammad1) [inline sd/StringParser/src/com/company/FileAdapter.java]: Throws what kind of exception?

File: sd/StringParser/src/com/company/StringFilter.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
+package com.company;
+import java.util.*;
+
+public class StringFilter {
+
+    public static String filter(String inputText, HashSet<String> blackList)
```
Comment (waqarmuhammad1) [inline sd/StringParser/src/com/company/StringFilter.java]: inputText is input string or input file name?

File: sd/StringParser/src/com/company/StringFilter.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    {
+        boolean firstPass = true;
+        Scanner scanner = new Scanner(inputText);
+        StringBuilder stringBuilder = new StringBuilder();
```
Comment (waqarmuhammad1) [inline sd/StringParser/src/com/company/StringFilter.java]: Could use StringJoiner sj = new StringJoiner(" "), in this case you wouldn't have to check for first pass

File: sd/StringParser/test/MainTest.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    public void tearDown() throws Exception {
+    }
+
+    @Test
+    public void mainTest() throws Exception {
+        File file = Paths.get("test\\tests\\cases.tsv").toFile();
+        Scanner scanner = new Scanner(file);
+        int currentLine = 0;
+
+        while(scanner.hasNextLine()) {
```
Comment (waqarmuhammad1) [inline sd/StringParser/test/MainTest.java]: block looks good, i didnt know that we can use scanner in this way too with delimiters

File: sd/StringParser/src/com/company/StringFilter.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+        // \s is a Java Pattern that represents all forms of white-space
+        scanner.useDelimiter("\\s");
+
+        while(scanner.hasNext())
+        {
+            String currentWord = scanner.next();
+            if (blackList.contains(currentWord.toUpperCase()))
```
Comment (waqarmuhammad1) [inline sd/StringParser/src/com/company/StringFilter.java]: what if blacklist word or input word has trailing spaces?

File: sd/StringParser/src/com/company/StringFilter.java (hunk 2 above)
Comment (seandough3rty) [reply sd/StringParser/src/com/company/StringFilter.java]: I didn't even think about that! good catch, refactored variable name to inputString

File: sd/StringParser/src/com/company/StringFilter.java [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+    public static String filter(String inputText, HashSet<String> blackList)
+    {
+        boolean firstPass = true;
+        Scanner scanner = new Scanner(inputText);
+        StringBuilder stringBuilder = new StringBuilder();
+
+        // \s is a Java Pattern that represents all forms of white-space
+        scanner.useDelimiter("\\s");
+
+        while(scanner.hasNext())
```
[the remark this answers is not in the document]
Comment (seandough3rty) [reply sd/StringParser/src/com/company/StringFilter.java]: Yes! I run each line of the test individually though, so the return value for the empty input test case would return an empty string. Or perhaps I'm not understanding correctly?

File: sd/StringParser/src/com/company/StringFilter.java (hunk 5 above)
Comment (seandough3rty) [reply sd/StringParser/src/com/company/StringFilter.java]: Good thought, this is solved by the delimiter thankfully

File: sd/StringParser/test/MainTest.java (hunk 4 above)
Comment (seandough3rty) [reply sd/StringParser/test/MainTest.java]: Thank you!

**Pull request closed without merging**